packages/ember-data/lib/serializers/rest_serializer.js

```diff
@@ -487,20 +490,22 @@ var RESTSerializer = JSONSerializer.extend({
     var payload = this.normalizePayload(rawPayload);
 
     for (var prop in payload) {
-      var typeName = this.typeForRoot(prop);
-      if (!store.modelFactoryFor(typeName, prop)) {
-        Ember.warn(this.warnMessageNoModelForKey(prop, typeName), false);
-        continue;
-      }
-      var type = store.modelFor(typeName);
-      var typeSerializer = store.serializerFor(type);
+      if (prop !== "meta") {
```

Comment:
Actually `extractMeta` mutates the original object passed in, by calling `delete payload.meta`, which means that the caller of the `pushPayload()` method, would be surprised to find that their payload object was mutated after calling the function.

The only way to avoid this is to either duplicate the payload object, then remove the key, or use the technique I use here to use comparisons to ignore the key. I think the original approach (which I did to avoid use of `delete`, which `extractMeta` uses) is the less of two evils. It's not ideal to run comparisons each iteration, but duplicating an object is even less ideal. 

Thoughts?